Odata-docs/client/v7/get-response-content.md

```diff
@@ -8,27 +8,74 @@ ms.date: 7/1/2019
+
+```csharp
+context.SaveChanges(SaveChangesOptions.BatchWithIndependentOperations);
+context.SaveChanges(SaveChangesOptions.BatchWithSingleChangeset);
+```
+
+A [DataServiceResponse](/dotnet/api/microsoft.odata.client.dataserviceresponse) object is returned after the `SaveChanges` operation is complete. The `DataServiceResponse` object includes a sequence of OperationResponse objects that, in turn, contain a sequence of `EntityDescriptor` or `LinkDescriptor` instances that represent the changes persisted or attempted. When an entity is created or modified in the data service, the EntityDescriptor includes a reference to the updated entity, including any server-generated property values, such as the generated ID value.
+
 When the service doesn't respond with `204 No Content` to data modification requests, the response contains a non-empty body. The code below helps to retrieve the body content:
 
 ``` csharp
 static void Main(string[] args)
 {
-    var context = new DefaultContainer(new Uri("https://services.odata.org/v4/(S(b0vguruqwzqbmfoanwq1guxc))/TripPinServiceRW/"));
+    var context = new DefaultContainer(new Uri("https://services.odata.org/v4/TripPinServiceRW/"));
 
     var person = Person.CreatePerson("russell", "Russell", "Whyte", new long());
 
     context.AddToPeople(person);
 
-    var responses = context.SaveChanges();
+    DataServiceResponse responses = context.SaveChanges();
 
-    foreach (var response in responses)
+    foreach (OperationResponse response in responses)
     {
         var changeResponse = (ChangeOperationResponse) response;
         var entityDescriptor = (EntityDescriptor) changeResponse.Descriptor;
         var personCreated = (Person) entityDescriptor.Entity; // the person created on the service
     }
 }
 ```
+The `DataServiceResponse` has the following properties that enable you to access additional information about a modification response:
+
+- [BatchHeaders](/dotnet/api/microsoft.odata.client.dataserviceresponse.batchheaders) - contains headers from an HTTP response associated with a batch request.
+- [BatchStatusCode](/dotnet/api/microsoft.odata.client.dataserviceresponse.batchstatuscode)- contains status code from an HTTP response associated with a batch request.
+- [IsBatchResponse](/dotnet/api/microsoft.odata.client.dataserviceresponse.isbatchresponse) - gets a Boolean value that indicates whether the response contains multiple results.
+
+The `DataServiceResponse` is enumerated to retrieve responses to operations being tracked by [OperationResponse](/dotnet/api/microsoft.odata.client.operationresponse) objects within the `DataServiceResponse`.
+
+## Query Responses
+We get a [QueryOperationResponse](/dotnet/api/microsoft.odata.client.operationresponse) after calling `Execute()` or `ExecuteAsync()` on `DataServiceQueries`.
```

Comment:
```suggestion
We get a [QueryOperationResponse](/dotnet/api/microsoft.odata.client.operationresponse) after calling `Execute()` or `ExecuteAsync()` on `DataServiceQuery`.
```

Reply:
If you apply code formatting, then it should be `DataServiceQuery` or `DataServiceQuery<T>` not `DataServiceQueries`

Reply:
Replaced it with `DataServiceQuery<TElement>`

Reply:
#Resolved